packages/poseidon-proof/src/poseidon-proof.circom

```diff
@@ -24,4 +24,6 @@ template PoseidonProof(NUMBER_OF_INPUTS) {
 
     // Dummy constraint to prevent compiler from optimizing it.
     signal dummySquare <== scope * scope;
+
+    dummySquare === scope * scope;
```

Comment:
https://github.com/trailofbits/circomspect/blob/main/doc/analysis_passes.md#under-constrained-signal

I don't get why this can be an issue. Is it clear to you?

Reply:
i have intended this as potentially "bad design" in the middle of the circuit logic but that's not our case. Definitely better to roll back to the previous intermediate unconstrained signal version - this adds no value.